src/energy.h

```diff
@@ -323,13 +323,59 @@ template <typename TSize, typename TSet> inline auto indexComplement(const TSize
     });
 }
 
+/**
+ * @brief A simple accumulator which immediately computes and adds energy of a pair of particles upon addition using
+ * the PairEnergy templated class.
+ *
+ * Generally this is the original way how the pairwise nonbonded energy has been computed in Faunus. Due to compiler
+ * optimization, templated class method 'PairEnergy.potential' may be inlined to significantly improve performance.
+ *
+ * @tparam TPairEnergy  pair energy implementing a potential(a, b) method for particles a and b
+ * @tparam TReduced  a primitive type to reduce into, e.g., double for energy
+ * @tparam TParticle
+ */
+template <typename TPairEnergy, typename TReduced, typename TParticle> struct EnergyAccumulator {
```

Comment:
May this should be renamed to `GenericEnergyAccumulator` or similar to reflect the summation method?

Reply:
Renamed to `BasicEnergyAccumulator`.